tests/integration/build_test.py

```diff
@@ -96,3 +96,16 @@ def test_build_with_dockerignore(self):
             list(filter(None, logs.split('\n'))),
             ['not-ignored'],
         )
+
+    @requires_api_version('1.9')
+    def test_build_with_buildargs(self):
+        script = io.BytesIO('\n'.join([
+            'FROM scratch',
+            'ARG test',
+            'USER $test'
+        ]).encode('ascii'))
+
+        self.client.build(fileobj=script, tag='buildargs',
+                          buildargs={'test': 'OK'}, stream=True)
```

Comment:
The created image needs to be cleaned up after the test has finished running. `self.tmp_imgs.append('buildargs')` should do the trick.